### PR TITLE
Account Awaiter

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

**Review by danielailie — APPROVED** [review]

File: multiversx_sdk/network_providers/account_awaiter_test.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from multiversx_sdk.network_providers.api_network_provider import \
+    ApiNetworkProvider
+from multiversx_sdk.network_providers.resources import AccountOnNetwork
+from multiversx_sdk.testutils.mock_network_provider import (
+    MockNetworkProvider, TimelinePointMarkCompleted, TimelinePointWait)
+from multiversx_sdk.testutils.utils import create_account_egld_balance
+from multiversx_sdk.testutils.wallets import load_wallets
+
+
+class TestTransactionAwaiter:
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/account_awaiter_test.py:16]: Typo / Account.

File: multiversx_sdk/network_providers/account_awaiter_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    provider = MockNetworkProvider()
+    watcher = AccountAwaiter(
+        fetcher=provider,
+        polling_interval_in_milliseconds=42,
+        timeout_interval_in_milliseconds=42 * 42,
+        patience_time_in_milliseconds=0
+    )
+
+    @pytest.mark.only
+    def test_await_on_balance_increase(self):
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/account_awaiter_test.py:26]: :rocket:

File: multiversx_sdk/network_providers/account_awaiter_test.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        )
+
+        def condition(account: AccountOnNetwork):
+            return account.balance == initial_balance + create_account_egld_balance(7)
+
+        account = self.watcher.await_on_condition(alice, condition)
+        assert account.balance == create_account_egld_balance(1007)
+
+    @pytest.mark.networkInteraction
+    def test_on_network(self):
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/account_awaiter_test.py:43]: Maybe say something about the balance increase in the test name? Like the one above, the one with the mock.

File: multiversx_sdk/network_providers/account_awaiter.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+logger = logging.getLogger("account_awaiter")
+
+
+class IAccountFetcher(Protocol):
+    def get_account(self, address: Address) -> AccountOnNetwork:
+        ...
+
+
+class AccountAwaiter:
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/account_awaiter.py:24]: Will we integrate it in the network providers in a next PR?

**Review by andreibancioiu — COMMENTED** [review]
:rocket:

File: multiversx_sdk/network_providers/account_awaiter_test.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk/network_providers/account_awaiter_test.py:16]: Fixed

File: multiversx_sdk/network_providers/account_awaiter_test.py (hunk 3 above)
Comment (popenta) [reply multiversx_sdk/network_providers/account_awaiter_test.py:43]: renamed to `test_await_for_account_balance_increase_on_network`.

File: multiversx_sdk/network_providers/account_awaiter.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk/network_providers/account_awaiter.py:24]: Forgot about it. Will do it in this PR. Thanks!

**Review by andreibancioiu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**